### ISSUE TITLE
The future of Headcoders in /tg/station13

### ISSUE THREAD

**Issue by RemieRichards** [tgstation/tgstation#30804]
Overly dramatic title yada yada.

| Maintainer | Cheridan          | Someone Else (Say Who) | Democracy of Maintainers (No Headcoder) |
|-------------|------------------|--------------------------------|:------------------------------------------------:|
| @AnturK  |
| @ChangelingRain | | optimumtact/oranges | :+1: 
| @Cyberboss  | | | :-1:
| @Jordie0608 |
| @lzimann | | lzimann/leo | :-1:
| @optimumtact |
| @Razharas |

Cheridan has proven time and time again he won't do anything with the position, and actively supports a frequently unwanted "contributor", should someone else take over? do we even need the position anymore?

In Cheridan's own words (from 2 years ago),
>Jalleo> what happened in the end with the idea of getting new headcoders?
Cheridan> if the code team wants to get more somehow i'm behind them 100%
Cheridan> i don't want to be the one deciding things there though
Cheridan> like i don't think it's a good political system for the guy in charge to be deciding elections
Cheridan> know what i'm saying?

Maintainers, vote.

This isn't about what I want, it's about what the team want.

**Comment by praisenarsie** [comment]
![photo](https://user-images.githubusercontent.com/31222036/30512213-0b6eb240-9ae2-11e7-87f0-96fcee4afd2f.jpg)

Question, how would no headcoder work exactly or will the maintainers have a discussion on how it would work?

Also, what if someone was voted to be headcoder yet refuses the position (for some reason)? I assume a revote?

also small suggestion, maybe limit this "issue"/conversation to collaborators/maintainers and put this "issue" on #coderbus's topic

**Comment by RemieRichards** [comment]
@praisenarsie as the column is labelled, "A Democracy of Maintainers" (more votes, essentially, just off-the-cuff ones in the bus, I imagine)

**Comment by kevinz000** [comment]
cyberboss for headcoder

**Comment by RemieRichards** [comment]
@kevinz000 your vote is appreciated, but only maintainers can vote.

**Comment by Cyberboss** [comment]
I don't plan on it but I'm wondering, are self votes allowed?

**Comment by Cyberboss** [comment]
Also, democracy of maintainers brings into question who gets admin on the
org

**Comment by DoctoreMad** [comment]
Pass the popcorn

**Comment by RemieRichards** [comment]
@Cyberboss a voted individual, I suppose. or perhaps @MrStonedOne

**Comment by RemieRichards** [comment]
@DoctoreMad I know this might seem amusing but I'm genuinely trying to change things for the better here.

**Comment by Cyberboss** [comment]
Also, Raz is occasionally active, does he not get a say?

**Comment by lzimann** [comment]
remie cant comment with this locked so i'll unlock and maybe delete irrelevant things

**Comment by RemieRichards** [comment]
Thanks @lzimann 

@Cyberboss 
If he wishes to, he can.
However he's:
A. inactive
B. private on the maintainer team meaning I couldn't see his name anymore and forgot to put him down.

**Comment by Cyberboss** [comment]
>PR owner cant comment on it

:thinking:

**Comment by RemieRichards** [comment]
Can't comment on it OR edit previous posts OR edit the Issue anymore.
Yes, it's incredibly dumb.

**Comment by praisenarsie** [comment]
just putting this here #11413

**Comment by DoctoreMad** [comment]
@RemieRichards No, it really isn't. If you were able to edit your previous posts people would just edit-argue after a mod locks the topic.
This isn't just in github, some forums work like that too. Its always incredibly special when it happens.

**Comment by RemieRichards** [comment]
@DoctoreMad I mean the issue author only.
I'm talking about myself in that post, no-one else.

**Comment by KorPhaeron** [comment]
Cheridan has kept the project stable for (4?) years now. I don't know if anyone else here can do that, especially if they're expected to be more involved in the day to day

**Comment by KorPhaeron** [comment]
What is "doing something?"

He fires/hires maintainers, he bans people from the rep, he rules on our arguments if we can't solve it with a vote. Other than banning goof, what do you want a head coder to be doing past that?

**Comment by RemieRichards** [comment]
"Hires/fires maintainers"
False, he always asks the maintainers, so that falls on the maintainers as a vote (as it should be), all he does is "give the keys", a purely ceremonial gesture

"bans people from the repo"
When was the last time he did this?

"he rules on our arguments"
No he doesn't.

**Comment by pubby** [comment]
I don't really understand what's going on, but I do know it was Cheridan who banned goof from doing his token memes.

**Comment by RemieRichards** [comment]
No, that was MSO and oranges, and it also didn't even fucking work because he did it again.

**Comment by KorPhaeron** [comment]
Last time I know of was when some hippie guys were raiding the repo several months back

**Comment by RemieRichards** [comment]
Oh yeah, sure, dealing with external cancers but not internal ones? he's so amazing

**Comment by lzimann** [comment]
I vote on myself(yes, I am serious).

**Comment by Repukan** [comment]
Stop being petty

**Comment by RemieRichards** [comment]
@Repukan your vote is appreciated, but only maintainers can vote.

**Comment by Repukan** [comment]
I'm not voting, I said stop acting like a kid.

You think you can run a project with how mentally unhinged you are/act?

This whole thing is a tantrum

**Comment by ike709** [comment]
I know only maintainers have a vote, but players still have an opinion, and I will voice mine whether you like it or not. A head coder that _actually does things_ (Don't defend Cheridan, I've been watching this GitHub repo almost constantly for over four years) could mean big changes for TG.

I still appreciate Remie's hard work on NTstation; and she has generally been a good maintainer here on TG. She was also nice and helpful on IRC back when I still used that. I support her for head coder.

Cyberboss, during his relatively short time here, has made several major refactors that broke the game, but in the process improved the overall code. From a code quality and maintainability standpoint, I support him for head coder.

Although oranges does a good job as a maintainer overall, he has closed pull requests numerous times just because he personally dislikes them. Even if the code is good and the community greatly supports it, he will close it. I'll admit, there were a few that were jokes, but the majority were genuine PRs. He fails to give a reason for closing the PRs beyond "I'm a maintainer." While it is within his power to close pull requests, he fails to provide valid reasons for the PRs he dislikes. Doing so on PRs that meet code quality standards and community approval should _not_ be allowed, and this is something I expect an active head coder to prevent. I do not support oranges for head coder, but I do support any maintainer who acknowledges his abuse of power on PRs he dislikes. This wouldn't be the case if he actually provided valid reasons for his actions on these PRs.

I don't support lzimann purely because they voted for themself. Yes, I am serious.

I won't comment on every maintainer, because this post is long enough already, but these are my top picks for/against head coder.

**Comment by RemieRichards** [comment]
>@Repukan your vote is appreciated, but only maintainers can vote.

**Comment by RemieRichards** [comment]
@ChangelingRain you can.

**Comment by Cyberboss** [comment]
As far as the democracy thing goes, I'm putting a :-1: for that.

My vote is still undecided

**Comment by Cyberboss** [comment]
As far as the voting goes, shouldn't we have IRV like the headmin elections?

**Comment by praisenarsie** [comment]
yes i'm not a maintainer but i'm just saying that self-voting is _kinda_ dumb

**Comment by RemieRichards** [comment]
That would require @MrStonedOne or something.
Which if its wanted, sure.

**Comment by Armhulen** [comment]
does this vote have any backing

**Comment by RemieRichards** [comment]
@Armhulen cheridan's previous commentary of being "100%" behind what the team want (in the OP, if you wish to read, also linked in full by praiseratvar), if the team voted him out, he would probably hold his word, if he's even an ounce of a respectable human being.

**Comment by DoctoreMad** [comment]
@RemieRichards  
You mean the two years old one
Did you double-check this

**Comment by RemieRichards** [comment]
@DoctoreMad Two years is a long time I admit yes, but Headcoder has always been a position where one steps down and passes it on.

Once it was finally removed from the clutches of whoever had it prior, it went to rockdtben (a very good headcoder actually), but he got a job and passed it on, unfortunately to cheridan.

If cheridan is voted out, and he refuses to leave, it just makes him look like another numbers or whatever, which i'm SURE will put him in a good light, totally.

**Comment by praisenarsie** [comment]
> unfortunately

why unfortunately?

**Comment by Armhulen** [comment]
narsie she explained it in op

**Comment by Armhulen** [comment]
i also kind of agree with remie on this somewhat ish i don't really care what you coders work out just work something out okay

**Comment by Cyberboss** [comment]
We can setup a third party IRV using https://civs.cs.cornell.edu/ it would require someone getting an email address from each maintainer

**Comment by KorPhaeron** [comment]
In all seriousness though I'm not sure why you get to decide whether I have a vote or not, you might dislike me or my job title but I am still an org member and do a significant amount of work reviewing PRs like everyone else here

**Comment by RemieRichards** [comment]
Your job title is insignificant, you are not a maintainer, the headcoder should be someone with actual code standards.

**Comment by pubby** [comment]
You are not a maintainer either, Remie, and have barely contributed to this game at all in the past few months.

**Comment by RemieRichards** [comment]
Having written/helped write a lot of the code standards we use today, that's irrelevant, my contributions continue without my having to do them personally.

**Comment by Armhulen** [comment]
that's why she isn't a vote either

but give kor a vote cmon now

**Comment by RemieRichards** [comment]
and also see that,  oh wait, I haven;'t voted? SHOCK HORROR it's almost as if I'm NOT being a hypocrite? wow!!

**Comment by RemieRichards** [comment]
and on that note @pubby when was the last time cheridan did?

**Comment by as334** [comment]
Based off of his github page, Cheridan's last action was merging a couple PRs in the first week of June and making an issue report.

**Comment by KorPhaeron** [comment]
I'm not intending to run for head coder, and the qualities you are seeking in a head coder have nothing to do with code standards, so I'm not sure how that's relevant. 

My job title has merge access the same as the maintainers and pretending that dealing with the most dramatic PRs is insignificant is petty.

**Comment by RemieRichards** [comment]
based on @pubby ignoring my reviewing of PRs, he clearly only considers actual PRs authored by that person as contribution.

And what a surprise, cheridan's last PR was... a few months ago! Looks like pubby's being a bit biased here!

**Comment by Cyberboss** [comment]
Kor gets a vote imo. He's both a maintainer and design lead.

**Comment by Armhulen** [comment]
give kor a vote

**Comment by RemieRichards** [comment]
@KorPhaeron Wjohn has merge access, your point is irrelevant.
You haven't been explicitly blocked from voting because I dislike how you are these days, or your "job title" you simply weren't included on the same grounds as everyone else wasn't, you're not a maintainer.
You lack code standards, and as such would have poor ability to choose someone who does, and if you seriously think a headcoder shouldn't have good code ability then you're not really helping your point.

@Cyberboss he is not, he is "design lead" he was never been a maintainer.

**Comment by DoctoreMad** [comment]
Wait a minute, did RemieRichards just set up the rules of this vote himself?

**Comment by RemieRichards** [comment]
@DoctoreMad first of all, Herself, second of all, I set up the rules as logically as possible for what is an impromptu and unofficial vote.

**Comment by RemieRichards** [comment]
Also nice edit for "Memie", very clever, shame that isn't even pronounced at all like the word you're trying to turn it into, or that it even makes a difference.

**Comment by DoctoreMad** [comment]
This vote is sudden and improptu as fuck, really.
Cheridan leaving is not guaranteed for reasons other than a 2 year old IRC message.
The vote rules are set up by someone who doesn't have enough power to consider [gendered pronoun] fit to vote. On the pool they created. Because well that's how we roll.
Considering it was proper enough to get 3 votes, I suppose I should shut up?

**Comment by RemieRichards** [comment]
Are you arguing with yourself there or what?

**Comment by pubby** [comment]
>and also see that, oh wait, I haven;'t voted?

Deciding who gets to vote is way more powerful than a single vote. Rejecting Kor and Cheridan is equivalent to two votes in your favor.

**Comment by Cyberboss** [comment]
@Cheridan has be the one deciding if kor, wj, remie, raz, or themself can vote, honestly. That's how the status quo stands.

**Comment by Militaires** [comment]
will this vote even change anything

**Comment by RemieRichards** [comment]
I can see that point of view, but cheridan rejected his own vote two years ago, or did you come here just to dramapost rather than read the op?

**Comment by RemieRichards** [comment]
@Cyberboss not a maintainer, not a maintainer, not a maintainer, maintainer away being a russian guardsman.

the distinction is clear.

**Comment by factoryman942** [comment]
I can feel the reddit posts of "tg on life support" already
👍

**Comment by Wyzack** [comment]
If this vote is entirely impromptu anyways cant Kor just declare who he wants to vote for and have it be considered just as valid as any other vote or not based on who is looking at the results?

**Comment by DoctoreMad** [comment]
@RemieRichards I will rewrite my post.
Your issue isn't really official, you didn't discuss the rules for the vote with other people(Cyberboss is a maintainer and thinks vote rules should be different, headmin Armhulen too), so this vote doesn't really have power. _Its more of a gentle suggestion_.
Does that make sense to you?

**Comment by RemieRichards** [comment]
@Wyzack
Well yes, of course.

@DoctoreMad yes it does, but considering i already said it was A. impromptu (as in, not discussed, sudden) and B. unofficial, I'm already well aware. I'm basing this on (admittedly a 2 year previous) version of cheridan who once claimed that they'd accept whatever the team decided, I'm merely the one who's poked the team to vote.

**Comment by Militaires** [comment]
don't accuse me of dramaposting while you started this remie.

second of all, why did you even need to publicize this?

**Comment by RemieRichards** [comment]
@Militaires I posted that to pubby, the post above, before you posted, github updated late.

It's public because I've quit and no longer have any way to handle this privately.

**Comment by Militaires** [comment]
very well.

**Comment by MrStonedOne** [comment]
my vote is that we blacklist remie until she reaches the point of maturity where she doesn't try to start a coup over the fact that she doesn't like goofball

**Comment by RemieRichards** [comment]
Says the weedlmao? That's totally mature of you.

**Comment by RemieRichards** [comment]
Also it was obvious how you were going to side, for some reason you offer goof these millions of chances too, but considering you actually DO SHIT for the community I didn't include you.

**Comment by Militaires** [comment]
remie just dissed recreational drugs, someone get colonic up in here

**Comment by DoctoreMad** [comment]
@MrStonedOne I don't care what Remie said I think this is what is passing through the heads of more than a few people.

**Comment by Arzion5** [comment]
![1503588939739](https://user-images.githubusercontent.com/21035676/30515823-caaa12a8-9b37-11e7-96d7-4c1a885f94c7.jpg)
Where is Goof?

**Comment by Militaires** [comment]
goof went into hiding after shooting both joan and remie in the asshole, he's a wanted man and we will find him for this crime

**Comment by praisenarsie** [comment]
goof is in ntrhut

**Comment by Arzion5** [comment]
![1503750440931](https://user-images.githubusercontent.com/21035676/30515834-2fb6b458-9b38-11e7-9d36-d0442a0b7689.png)

**Comment by Wyzack** [comment]
To be entirely honest after goof didnt get repobanned for the whole goatse thing i think we all assumed he was unbannable

**Comment by Firecage** [comment]
@MrStonedOne Well, goofball HAS had millions of chances, yet he still keeps fucking up. And it doesn't matter whether it is on IRC, Github, the server, etc., he always acts toxic as fuck. Holy fuck, he has like 480 closed PR's compared to 218 or so merged PR's. The dude either makes a bunch of meme PR's, joke PR's, or PR's which are just plain horribly coded. Holy fuck, many times he even refuses to listen to the most basic of feedback/criticism unless it is forced down his throat. Bloody hell, he even refuses to test his own shit many times, nor does he bother fixing his own features to often.

Heck, ONCE AGAIN he was even blatantly bribing people for votes during the last headmin election...like he has done every single headmin election he has taken part in.

So MSO, whether this is about goof or not. The fact that he is still around is not a good thing at all. Anyone else would have been given the pink slip long ago, and booted straight into a bucket. Yet somehow it appears that guy is unbannable and very, very rarely gets a slap on the wrist.

**Comment by ChangelingRain** [comment]
This drama isn't drama enough I'm changing my vote to @Iamgoofball

**Comment by RemieRichards** [comment]
I lol'd.

**Comment by DoctoreMad** [comment]
Oh yeah we should wait until CR is serious.
First he voted Orange after someone denounced Oranges and now he's switching to Goofball

**Comment by Militaires** [comment]
do you guys see why i support goof every time now

this is what he does

**Comment by RemieRichards** [comment]
She won't ever be serious about any of this, she's dealing with her own shit at the moment.

**Comment by RemieRichards** [comment]
@Militaires but if he was actually removed, none of this would happen, so if you truely only support him to silence drama (that he creates...) then that's a very weird reason.

**Comment by GunHog** [comment]
Clearly some people just want to watch the codebase burn.

**Comment by DoctoreMad** [comment]
What's with all those women and their awful opinions

**Comment by InsaneHyena** [comment]
No, none of this would happen if you stopped being hysterical.

**Comment by Qustinnus** [comment]
wasn't this about catpeople or something. i knew they'd ruin TG, but maybe not in this way.

**Comment by AuroranAI** [comment]
the slippery slope is here and someone's cut the brakes

**Comment by Militaires** [comment]
god fucking dammit this is so fucking funny

who needs tv when you have this

**Comment by Qustinnus** [comment]
@RemieRichards unblock Remie so he can discuss in this thread, as they are involved.

**Comment by The-clown-known-as-Pepe** [comment]
@Qustinnus dumb

**Comment by RemieRichards** [comment]
@Qustinnus ? what?

**Comment by The-clown-known-as-Pepe** [comment]
@RemieRichards he's dumb excuse him

**Comment by AuroranAI** [comment]
the s m a r t e s t boy

**Comment by The-clown-known-as-Pepe** [comment]
he traps for gold in runescape

**Comment by Qustinnus** [comment]
I ment goof, I have autism pls excuse

**Comment by The-clown-known-as-Pepe** [comment]
wts rune dagger 15m

**Comment by RussellTalbott** [comment]
https://i.imgur.com/uMt37yz.png

**Comment by Qustinnus** [comment]
@The-clown-known-as-Pepe i need to get my bonds somehow

**Comment by RemieRichards** [comment]
@Qustinnus I waited to comment after you corrected yourself, I don't know what you're talking about?
Is he posting here or something? because I've had him blocked for a lot longer than this shit, thought it only hid comments like other things.

**Comment by Qustinnus** [comment]
He can't post on your issues if he's blocked.

**Comment by DoctoreMad** [comment]
Okay first things first, who linked all those TG derivatives here? This was better when people were pretending to be serious, then comes reddit-lite to try to be ironic.

As for @RemieRichards ,you should unblock goof not because of justice or anything but because he will shit the bed.

@Qustinnus Remie may or may not have a grudge because of goof messing with catgirls, but that's really unrelated.

**Comment by dannno** [comment]
i miss HG :[

**Comment by RemieRichards** [comment]
@Qustinnus Hilarious.
I'm not sure how to act on that one.

His opinion is already scrawled everywhere he can put it (forums, irc, reddit, in game I imagine) so I'm not sure what more allowing him to post here would do.

**Comment by The-clown-known-as-Pepe** [comment]
@DoctoreMad who are you again?

**Comment by dannno** [comment]
![](https://i.imgur.com/qRozXgM.png)

**Comment by DoctoreMad** [comment]
@The-clown-known-as-Pepe Not important really, what is important is what I'm going to do.

**Comment by AuroranAI** [comment]
i'm crashing this issue with no survivors

**Comment by The-clown-known-as-Pepe** [comment]
@DoctoreMad what are you going to do?

**Comment by dannno** [comment]
im downaloding 40 gigabites of snake pussy and ass and its making it really hard to  loa d thi  s page

**Comment by Militaires** [comment]
@dannno can u send me the file

**Comment by The-clown-known-as-Pepe** [comment]
@dannno i thought you were more of a cat boy

**Comment by The-clown-known-as-Pepe** [comment]
Also include me in the reddit screencap please.

**Comment by Qustinnus** [comment]
@hornygranny help us

**Comment by Slignerd** [comment]
meow?

**Comment by dannno** [comment]
HG FOR HEADGODER

**Comment by The-clown-known-as-Pepe** [comment]
@Slignerd nya~?

**Comment by AuroranAI** [comment]
even the catgirl poster who was the #1 person against goof's removal of catgirls is shitposting here
the end is nigh

**Comment by dannno** [comment]
sauce me up some of that PUSSy

**Comment by dannno** [comment]
👏 its 👏 HG's 👏 TURN

**Comment by dannno** [comment]
👏

**Comment by dannno** [comment]
anyone want to party up; with us in Dota of the Ancients 2

**Comment by Armhulen** [comment]
no im launching one way heroics

**Comment by Armhulen** [comment]
pick a class for me 2 play tho ok

**Comment by Qustinnus** [comment]
dude warcraft 3 DOTA or get out

**Comment by Armhulen** [comment]
this post made me frown

**Comment by dannno** [comment]
faggot way heroices you fuckign faggot you are so fucking stupid gay faggot game

**Comment by lzimann** [comment]
i was going to clean this up but github takes forever so i'll just keep this locked

**Comment by DoctoreMad** [comment]
Can the guys that """Can't vote""" please vote? The only thing that will happen is that RemieRichards won't include you in the issue OP, she can't actually decide what votes will be accepted.

**Comment by RemieRichards** [comment]
If they really wanted to it's surprising they already haven't.

**Comment by Repukan** [comment]
Add me to the screenshot

**Comment by optimumtact** [comment]
Cheridan has my full support, and as far as I'm concerned you're making this issue to cause drama, so you can stop.

**Comment by optimumtact** [comment]
Basically none of the people commenting here have any idea of the extent of the work Cheridan does, ceremonial or otherwise, Especially people like @ike709 who should stay on their own side of the fence

**Comment by optimumtact** [comment]
I I have spoken with Remie and amended my comment about them not being in IRC as they have been I just did not interact or notice.
